flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/PerJobCheckpointRecoveryFactory.java

```diff
@@ -38,7 +41,7 @@
     public static <T extends CompletedCheckpointStore>
             CheckpointRecoveryFactory withoutCheckpointStoreRecovery(IntFunction<T> storeFn) {
         return new PerJobCheckpointRecoveryFactory<>(
-                (maxCheckpoints, previous) -> {
+                (maxCheckpoints, previous, sharedStateRegistry) -> {
```

Comment:
Why do you need this extra parameter? You have the `sharedStateRegistry` inside of the `previous` already.

Reply:
I guess `previous` is optional during the 1st invokation?